students/ShujinCao/lesson02/assignment/code/charges_calc.py

```diff
@@ -0,0 +1,116 @@
+import logging
+
+def main():
+    """
+    main function
+    """
+    log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+    log_file = datetime.datetime.now().strftime("%Y-%m-%d") + ".log"
+    args = parse_cmd_arguments()
+    if args.debug == 1:
+        logging.basicConfig(level=logging.ERROR, format=log_format,
+                            filename=log_file, filemode="a+")
+    if args.debug == 2:
+        logging.basicConfig(level=logging.WARNING, format=log_format,
+                            filename=log_file, filemode="a+")
+    if args.debug == 3:
+        logging.basicConfig(level=logging.DEBUG, format=log_format,
+                            filename=log_file, filemode="a+")
+    input_data = load_rentals_file(args.input)
+    logging.warning("DATA READ IN DONE - CALCULATION START")
+    output_data = calculate_additional_fields(input_data)
+    logging.warning("CALCULATION DONE - SAVE OUTPUT TO JSON FILE")
+    save_to_json(args.output, output_data)
+    logging.warning("OUTPUT HAS BEEN SUCCESFULLY SAVED")
+
+def parse_cmd_arguments():
+    """
+    add arguments
+    """
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument(
+        '-o',
+        '--output',
+        help='ouput JSON file',
+        required=True)
+    parser.add_argument(
+        "-d",
+        "--debug",
+        type=int,
+        choices=range(
+            0,
+            4),
+        help="0: No debug messages or log file.\
+                         1: Only error messages.\
+                         2: Error messages and warnings.\
+                         3: Error messages, warnings and debug messages")
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    load json file
+    """
+    with open(filename) as file:
+        try:
+            json_f = json.load(file)
+        except BaseException:
+            logging.error("DATA WAS NOT LOADED.")
+            exit(0)
+    return json_f
+
+
+def calculate_additional_fields(data):
+    """
+    calculations
+    """
+    for value in data.values():
+        rental_start = datetime.datetime.strptime(
+            value['rental_start'], '%m/%d/%y')
+        try:
+            rental_end = datetime.datetime.strptime(
+                value['rental_end'], '%m/%d/%y')
+        except ValueError:
+            logging.error("CALCULATION ERROR FOR RENTAL_END")
+            logging.warning(
+                "QUICK FIX : REPLACING RENTAL_END WITH RENTAL_START")
+            rental_end = datetime.datetime.strptime(
+                value['rental_start'], '%m/%d/%y')
+        logging.debug("rental_start : %s; rental_end : %s", rental_start, rental_end)
+        value['total_days'] = (rental_end - rental_start).days
+        value['total_price'] = abs(
+            value['total_days'] *
+            value['price_per_day'])
```

Comment:
Notice that taking the absolute value of the total price, you are making an assumption about the data. Always verify your assumptions with the customer before implementing them in code.